unit/solvers/smt2_incremental/smt_index.cpp

```diff
@@ -0,0 +1,69 @@
+// Author: Diffblue Ltd.
+
+#include <util/optional.h>
+
+#include <solvers/smt2_incremental/smt_index.h>
+#include <testing-utils/use_catch.h>
+
+TEST_CASE("Test smt_indext.pretty is accessible.", "[core][smt2_incremental]")
+{
+  const smt_indext index = smt_numeral_indext{42};
+  REQUIRE(index.pretty(0, 0) == "smt_numeral_index\n  * value: 42");
+}
+
+TEST_CASE("Test smt_index getters", "[core][smt2_incremental]")
+{
+  SECTION("Numeral")
+  {
+    REQUIRE(smt_numeral_indext{42}.value() == 42);
+  }
+  SECTION("Symbol")
+  {
+    REQUIRE(smt_symbol_indext{"foo"}.identifier() == "foo");
```

Comment:
What happens if you create a `smt_symbol_indext{""}`?

Reply:
Then the constructed instant will contain the empty string. Are you suggesting an additional invariant?